server/src/chaos/models.rs

```diff
@@ -1,41 +1,588 @@
+
+                // clear all set related to disruption
+                self.impacts_set.clear();
+                self.tags_set.clear();
+                self.properties_set.clear();
+                self.impact_object_set.clear();
+
+                self.disruptions.push(disruption);
+                let idx: usize = self.disruptions.len() - 1;
+                entry.insert(idx);
+                self.disruptions.last_mut().unwrap()
+            }
+            Occupied(entry) => self.disruptions.get_mut(*entry.get()).unwrap(),
+        };
+        DisruptionMaker::update_tags(&mut self.tags_set, row, disruption)?;
+        DisruptionMaker::update_properties(&mut self.properties_set, row, disruption)?;
+        DisruptionMaker::update_impacts(
+            &mut self.impact_object_set,
+            &mut self.impacts_set,
+            row,
+            disruption,
+        )?;
+        Ok(())
+    }
+
+    fn make_disruption(row: &ChaosDisruption) -> Result<chaos_proto::chaos::Disruption, Error> {
+        let mut disruption = chaos_proto::chaos::Disruption::new();
+        disruption.set_id(row.disruption_id.to_string());
+        disruption.set_contributor(row.contributor.clone());
+        if let Some(reference) = &row.disruption_reference {
+            disruption.set_reference(reference.clone());
+        }
+        disruption.set_created_at(u64::try_from(row.disruption_created_at.timestamp())?);
+        if let Some(updated_at) = &row.disruption_updated_at {
+            disruption.set_updated_at(u64::try_from(updated_at.timestamp())?);
+        }
+        // Fill cause
+        let cause = disruption.mut_cause();
+        cause.set_wording(row.cause_wording.clone());
+        if let Some(category_name) = &row.category_name {
+            let category = cause.mut_category();
+            category.set_name(category_name.clone());
+        }
+        // Fill publication_period
+        let publication_period = disruption.mut_publication_period();
+        if let Some(start) = &row.disruption_start_publication_date {
+            publication_period.set_start(u64::try_from(start.timestamp())?);
+        }
+        if let Some(end) = &row.disruption_end_publication_date {
+            publication_period.set_end(u64::try_from(end.timestamp())?);
+        }
+        Ok(disruption)
+    }
+
+    fn update_tags(
+        tags_set: &mut HashSet<Uid>,
+        row: &ChaosDisruption,
+        disruption: &mut chaos_proto::chaos::Disruption,
+    ) -> Result<(), Error> {
+        if let Some(tag_id) = row.tag_id {
+            if !tags_set.contains(&tag_id) {
+                let mut tag = chaos_proto::chaos::Tag::new();
+                tag.set_id(tag_id.to_string());
+                if let Some(name) = &row.tag_name {
+                    tag.set_name(name.clone());
+                }
+
+                disruption.tags.push(tag);
+                tags_set.insert(tag_id);
+            }
```

Comment:
```suggestion
                let mut tag = chaos_proto::chaos::Tag::new();
                tag.set_id(tag_id.to_string());
                if let Some(name) = &row.tag_name {
                    tag.set_name(name.clone());
                }

                disruption.tags.push(tag);
                tags_set.insert(tag_id);
            
```
that's a HashSet, it won't insert a duplicate element